src/node/types.h

```diff
@@ -43,6 +44,10 @@ struct BlockCreateOptions {
      * transaction outputs.
      */
     size_t coinbase_output_max_additional_sigops{400};
+    /**
+     * Script to put in the coinbase transaction.
+     */
+    std::optional<CScript> coinbase_script{};
```

Comment:
In production this is always a dummy. `CreateNewBlock()` sets the script to `OP_TRUE` and then the `getblocktemplate` RPC omits the whole coinbase transaction.

It's not safe for the caller to set an _arbitrary_ dummy, because IIUC if it's larger than `coinbase_max_additional_weight` the `TestBlockValidity` internal check for `getblocktemplate` will spuriously fail. Which is why I think it's better for `CreateNewBlock` to decide what the dummy should be.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/31318#discussion_r1846813317

Would it make sense to declare the field as `CScript coinbase_output_script{CScript{} << OP_TRUE}` instead of `std::optional<CScript> coinbase_output_script{};` just to be clearer about what the default value is? Or maybe just add documentation to say what happens if this is not set?

I think the drawback of using std::optional here is that it's not really clear what happens if the option is not set. It's also not clear when you would want to set the option here, if ever.

Reply:
I expanded the comment to explain things a bit more.

I think you should not set this option unless you care what the coinbase output is. For example the `block_assemble` benchmark uses `P2WSH_OP_TRUE`. I'm guessing that's because it wants to look at the performance related to spending from a segwit coinbase.